include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h

```diff
@@ -34,19 +35,20 @@ namespace dpl
 namespace __par_backend_hetero
 {
 
-struct __leaf_sort_kernel
+struct __subgroup_bubble_sorter
 {
-    template <typename _Acc, typename _Size1, typename _Compare>
+    template <typename _Storage, typename _Compare, typename _Size>
     void
-    operator()(const _Acc& __acc, const _Size1 __start, const _Size1 __end, _Compare __comp) const
+    sort(_Storage& __storage, _Compare __comp, _Size __start, _Size __end) const
     {
-        for (_Size1 i = __start; i < __end; ++i)
+        for (_Size i = __start; i < __end; ++i)
         {
-            for (_Size1 j = __start + 1; j < __start + __end - i; ++j)
+            for (_Size j = __start + 1; j < __start + __end - i; ++j)
             {
                 // forwarding references allow binding of internal tuple of references with rvalue
-                auto&& __first_item = __acc[j - 1];
-                auto&& __second_item = __acc[j];
+                // TODO: avoid bank conflicts, or reconsider the algorithm
+                auto&& __first_item = __storage[j - 1];
```

Comment:
As we may see at https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_interface_for_local_accessors, `operator[]` return reference:
```C++
  /* Available only when: (Dimensions > 0) */
  reference operator[](id<Dimensions> index) const;
```

So why you are using `&&` here ?

Reply:
Indeed, forwarding reference is not needed with the current changes. I've removed it.